packages/svelte/src/lib/types.ts

```diff
@@ -1,10 +1,10 @@
 import type { PageProps } from '@inertiajs/core'
 import type { ComponentType } from 'svelte'
 
-export type ComponentResolver = (name: string) => ComponentType | Promise<ComponentType>
+export type ComponentsResolver = (name: string) => ResolvedComponents | Promise<ResolvedComponents>
 
-export interface InertiaComponentType extends ComponentType {
-  default: InertiaComponentType
-  layout: InertiaComponentType
-  props: PageProps
+export type ResolvedComponents = {
```

Comment:
This should also be singular, yes? `ResolvedComponent`?

Reply:
@joetannenbaum  It would be plural if the layout prop is filled. I do not mind making them singular as it d

Reply:
It's done ! :)